#
# Node parameters.
#

camera_topic: '/cv_camera_node/image_raw'
#camera_topic: '/maeve_video_file/image_raw'
#camera_topic: '/raspicam_node/image_rect'
#camera_topic: '/raspicam_node/image'
image_transport: 'compressed' # uncomment this to listen to image_transport topics

# node-relative topic, leave empty to disable visualization
viz_isp_field_topic: ''
#viz_isp_field_topic: 'viz_isp_field'

# node-relative topic for control command output
control_command_topic: 'control_command'

# bounds for scaling potential values for visualization; must be two elements with first less than zero and second greater than zero
# it makes sense for the lower bound to be equal to the range_min of soft constraint transform
viz_potential_bounds: [-10.0, 10.0] 

# Print output to terminal during exectution?
verbose: false

#
# Image Space Potential field parameters
#

hard_constraint_transform:
  alpha: 1.0
  beta: 1.0
  range_min: 0.0
  range_max: 0.5

soft_constraint_transform:
  alpha: 1.0
  beta: 1.0
  range_min: 5.0
  range_max: 10.0

#
# AR tag tracking parameters
#

ar_tag_max_age: 0.5 # must be >= 0.0
ar_time_queue_size: 10 # number of elements to allocate for circular buffer
ar_time_queue_max_gap: 0.5 # seconds
output_frame_param_name: '/ar_track_alvar/output_frame'
marker_size_param_name: '/ar_track_alvar/marker_size'
ar_frame_prefix: 'ar_marker_'

# these need to match the set of tags being tracked by ar_track_alvar
ar_tag_obstacle_ids: [0] #, 1, 2, 3, 4, 5, 6, 7, 8]
ar_tag_target_ids: []

#
# Visual servoing parameters
#

isp_controller_params:
  kernel_width: 3
  kernel_height: 3
  kernel_horizon: 0.5 # \in [0, 1] with 0 - top and 1 - bottom of image
  yaw_decay_left: 1.05 # \TODO(me) probably don't call this 'decay'
  yaw_decay_right: 1.05 # \TODO(me) probably don't call this 'decay'
  K_P: 1.0
  K_D: 0.0
  potential_inertia: 0.0
  shape_parameters:
    range_min: -1.0
    range_max: 1.0
    alpha: 0.05
    beta: 0.5
